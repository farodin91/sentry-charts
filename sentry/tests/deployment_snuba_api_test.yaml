suite: test deployment-snuba-api
templates:
  - configmap-snuba.yaml
  - deployment-snuba-api.yaml
tests:
  - it: update image
    template: deployment-snuba-api.yaml
    set:
      images.snuba.tag: test
    asserts:
      - isKind:
          of: Deployment
      - matchRegex:
          path: metadata.name
          pattern: -sentry-snuba-api$
      - equal:
          path: spec.template.spec.containers[0].image
          value: getsentry/snuba:test
  - it: update dnsPolicy
    template: deployment-snuba-api.yaml
    set:
      dnsPolicy: test
    asserts:
      - isKind:
          of: Deployment
      - matchRegex:
          path: metadata.name
          pattern: -sentry-snuba-api$
      - equal:
          path: spec.template.spec.dnsPolicy
          value: test
  - it: update revisionHistoryLimit
    template: deployment-snuba-api.yaml
    set:
      revisionHistoryLimit: 3
    asserts:
      - isKind:
          of: Deployment
      - matchRegex:
          path: metadata.name
          pattern: -sentry-snuba-api$
      - equal:
          path: spec.revisionHistoryLimit
          value: 3
  - it: set replicas
    template: deployment-snuba-api.yaml
    set:
      snuba.api.replicas: 5
    asserts:
      - isKind:
          of: Deployment
      - matchRegex:
          path: metadata.name
          pattern: -sentry-snuba-api$
      - equal:
          path: spec.replicas
          value: 5
  - it: set role label
    template: deployment-snuba-api.yaml
    asserts:
      - isKind:
          of: Deployment
      - matchRegex:
          path: metadata.name
          pattern: -sentry-snuba-api$
      - equal:
          path: spec.template.metadata.labels.role
          value: snuba-api
      - equal:
          path: spec.selector.matchLabels.role
          value: snuba-api
  - it: set podlabels
    template: deployment-snuba-api.yaml
    set:
      snuba.api.podLabels:
        test: test
    asserts:
      - isKind:
          of: Deployment
      - matchRegex:
          path: metadata.name
          pattern: -sentry-snuba-api$
      - equal:
          path: spec.template.metadata.labels.test
          value: test
  - it: set sidecars
    template: deployment-snuba-api.yaml
    set:
      snuba.api.sidecars:
        - name: test
          image: nginx:latest
    asserts:
      - isKind:
          of: Deployment
      - matchRegex:
          path: metadata.name
          pattern: -sentry-snuba-api$
      - equal:
          path: spec.template.spec.containers[1].image
          value: nginx:latest
  - it: set resources
    template: deployment-snuba-api.yaml
    set:
      snuba.api.resources:
        limits:
          memory: 64Mi
    asserts:
      - isKind:
          of: Deployment
      - matchRegex:
          path: metadata.name
          pattern: -sentry-snuba-api$
      - equal:
          path: spec.template.spec.containers[0].resources.limits.memory
          value: 64Mi
  - it: set volumeMounts
    template: deployment-snuba-api.yaml
    set:
      snuba.api.volumeMounts:
        - mountPath: /etc/bla
          name: bla
    asserts:
      - isKind:
          of: Deployment
      - matchRegex:
          path: metadata.name
          pattern: -sentry-snuba-api$
      - equal:
          path: spec.template.spec.containers[0].volumeMounts[0].name
          value: config
      - equal:
          path: spec.template.spec.containers[0].volumeMounts[1].name
          value: bla
          
          