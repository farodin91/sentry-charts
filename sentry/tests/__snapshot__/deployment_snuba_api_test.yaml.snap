set podlabels:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: sentry
        chart: sentry-19.2.0
        heritage: Helm
        release: RELEASE-NAME
      name: RELEASE-NAME-sentry-snuba-api
    spec:
      replicas: 1
      revisionHistoryLimit: 10
      selector:
        matchLabels:
          app: RELEASE-NAME-sentry
          release: RELEASE-NAME
          role: snuba-api
      template:
        metadata:
          annotations:
            checksum/config.yaml: 23f36c63e6ee38f6cd97e6889384bb53bc60cbfe742fd1c202f24d275363e850
            checksum/snubaSettingsPy: d5f85a6a8afbc55eebe23801e1a51a0fb4c0428c9a73ef6708d8dc83e079cd49
          labels:
            app: RELEASE-NAME-sentry
            release: RELEASE-NAME
            role: snuba-api
            test: test
        spec:
          containers:
            - env:
                - name: SNUBA_SETTINGS
                  value: /etc/snuba/settings.py
                - name: DEFAULT_BROKERS
                  value: RELEASE-NAME-kafka:9092
              envFrom:
                - secretRef:
                    name: RELEASE-NAME-sentry-snuba-env
              image: getsentry/snuba:23.5.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /
                  port: 1218
                  scheme: HTTP
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 2
              name: sentry-snuba
              ports:
                - containerPort: 1218
              readinessProbe:
                failureThreshold: 10
                httpGet:
                  path: /
                  port: 1218
                  scheme: HTTP
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 2
              volumeMounts:
                - mountPath: /etc/snuba
                  name: config
                  readOnly: true
          volumes:
            - configMap:
                name: RELEASE-NAME-sentry-snuba
              name: config
set replicas:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: sentry
        chart: sentry-19.2.0
        heritage: Helm
        release: RELEASE-NAME
      name: RELEASE-NAME-sentry-snuba-api
    spec:
      replicas: 5
      revisionHistoryLimit: 10
      selector:
        matchLabels:
          app: RELEASE-NAME-sentry
          release: RELEASE-NAME
          role: snuba-api
      template:
        metadata:
          annotations:
            checksum/config.yaml: 23f36c63e6ee38f6cd97e6889384bb53bc60cbfe742fd1c202f24d275363e850
            checksum/snubaSettingsPy: d5f85a6a8afbc55eebe23801e1a51a0fb4c0428c9a73ef6708d8dc83e079cd49
          labels:
            app: RELEASE-NAME-sentry
            release: RELEASE-NAME
            role: snuba-api
        spec:
          containers:
            - env:
                - name: SNUBA_SETTINGS
                  value: /etc/snuba/settings.py
                - name: DEFAULT_BROKERS
                  value: RELEASE-NAME-kafka:9092
              envFrom:
                - secretRef:
                    name: RELEASE-NAME-sentry-snuba-env
              image: getsentry/snuba:23.5.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /
                  port: 1218
                  scheme: HTTP
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 2
              name: sentry-snuba
              ports:
                - containerPort: 1218
              readinessProbe:
                failureThreshold: 10
                httpGet:
                  path: /
                  port: 1218
                  scheme: HTTP
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 2
              volumeMounts:
                - mountPath: /etc/snuba
                  name: config
                  readOnly: true
          volumes:
            - configMap:
                name: RELEASE-NAME-sentry-snuba
              name: config
set resources:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: sentry
        chart: sentry-19.2.0
        heritage: Helm
        release: RELEASE-NAME
      name: RELEASE-NAME-sentry-snuba-api
    spec:
      replicas: 1
      revisionHistoryLimit: 10
      selector:
        matchLabels:
          app: RELEASE-NAME-sentry
          release: RELEASE-NAME
          role: snuba-api
      template:
        metadata:
          annotations:
            checksum/config.yaml: 23f36c63e6ee38f6cd97e6889384bb53bc60cbfe742fd1c202f24d275363e850
            checksum/snubaSettingsPy: d5f85a6a8afbc55eebe23801e1a51a0fb4c0428c9a73ef6708d8dc83e079cd49
          labels:
            app: RELEASE-NAME-sentry
            release: RELEASE-NAME
            role: snuba-api
        spec:
          containers:
            - env:
                - name: SNUBA_SETTINGS
                  value: /etc/snuba/settings.py
                - name: DEFAULT_BROKERS
                  value: RELEASE-NAME-kafka:9092
              envFrom:
                - secretRef:
                    name: RELEASE-NAME-sentry-snuba-env
              image: getsentry/snuba:23.5.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /
                  port: 1218
                  scheme: HTTP
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 2
              name: sentry-snuba
              ports:
                - containerPort: 1218
              readinessProbe:
                failureThreshold: 10
                httpGet:
                  path: /
                  port: 1218
                  scheme: HTTP
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 2
              resources:
                limits:
                  memory: 64Mi
              volumeMounts:
                - mountPath: /etc/snuba
                  name: config
                  readOnly: true
          volumes:
            - configMap:
                name: RELEASE-NAME-sentry-snuba
              name: config
set sidecars:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: sentry
        chart: sentry-19.2.0
        heritage: Helm
        release: RELEASE-NAME
      name: RELEASE-NAME-sentry-snuba-api
    spec:
      replicas: 1
      revisionHistoryLimit: 10
      selector:
        matchLabels:
          app: RELEASE-NAME-sentry
          release: RELEASE-NAME
          role: snuba-api
      template:
        metadata:
          annotations:
            checksum/config.yaml: 23f36c63e6ee38f6cd97e6889384bb53bc60cbfe742fd1c202f24d275363e850
            checksum/snubaSettingsPy: d5f85a6a8afbc55eebe23801e1a51a0fb4c0428c9a73ef6708d8dc83e079cd49
          labels:
            app: RELEASE-NAME-sentry
            release: RELEASE-NAME
            role: snuba-api
        spec:
          containers:
            - env:
                - name: SNUBA_SETTINGS
                  value: /etc/snuba/settings.py
                - name: DEFAULT_BROKERS
                  value: RELEASE-NAME-kafka:9092
              envFrom:
                - secretRef:
                    name: RELEASE-NAME-sentry-snuba-env
              image: getsentry/snuba:23.5.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /
                  port: 1218
                  scheme: HTTP
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 2
              name: sentry-snuba
              ports:
                - containerPort: 1218
              readinessProbe:
                failureThreshold: 10
                httpGet:
                  path: /
                  port: 1218
                  scheme: HTTP
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 2
              volumeMounts:
                - mountPath: /etc/snuba
                  name: config
                  readOnly: true
            - image: nginx:latest
              name: test
          volumes:
            - configMap:
                name: RELEASE-NAME-sentry-snuba
              name: config
set volumeMounts:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: sentry
        chart: sentry-19.2.0
        heritage: Helm
        release: RELEASE-NAME
      name: RELEASE-NAME-sentry-snuba-api
    spec:
      replicas: 1
      revisionHistoryLimit: 10
      selector:
        matchLabels:
          app: RELEASE-NAME-sentry
          release: RELEASE-NAME
          role: snuba-api
      template:
        metadata:
          annotations:
            checksum/config.yaml: 23f36c63e6ee38f6cd97e6889384bb53bc60cbfe742fd1c202f24d275363e850
            checksum/snubaSettingsPy: d5f85a6a8afbc55eebe23801e1a51a0fb4c0428c9a73ef6708d8dc83e079cd49
          labels:
            app: RELEASE-NAME-sentry
            release: RELEASE-NAME
            role: snuba-api
        spec:
          containers:
            - env:
                - name: SNUBA_SETTINGS
                  value: /etc/snuba/settings.py
                - name: DEFAULT_BROKERS
                  value: RELEASE-NAME-kafka:9092
              envFrom:
                - secretRef:
                    name: RELEASE-NAME-sentry-snuba-env
              image: getsentry/snuba:23.5.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /
                  port: 1218
                  scheme: HTTP
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 2
              name: sentry-snuba
              ports:
                - containerPort: 1218
              readinessProbe:
                failureThreshold: 10
                httpGet:
                  path: /
                  port: 1218
                  scheme: HTTP
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 2
              volumeMounts:
                - mountPath: /etc/snuba
                  name: config
                  readOnly: true
                - mountPath: /etc/bla
                  name: bla
          volumes:
            - configMap:
                name: RELEASE-NAME-sentry-snuba
              name: config
update dnsPolicy:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: sentry
        chart: sentry-19.2.0
        heritage: Helm
        release: RELEASE-NAME
      name: RELEASE-NAME-sentry-snuba-api
    spec:
      replicas: 1
      revisionHistoryLimit: 10
      selector:
        matchLabels:
          app: RELEASE-NAME-sentry
          release: RELEASE-NAME
          role: snuba-api
      template:
        metadata:
          annotations:
            checksum/config.yaml: 23f36c63e6ee38f6cd97e6889384bb53bc60cbfe742fd1c202f24d275363e850
            checksum/snubaSettingsPy: d5f85a6a8afbc55eebe23801e1a51a0fb4c0428c9a73ef6708d8dc83e079cd49
          labels:
            app: RELEASE-NAME-sentry
            release: RELEASE-NAME
            role: snuba-api
        spec:
          containers:
            - env:
                - name: SNUBA_SETTINGS
                  value: /etc/snuba/settings.py
                - name: DEFAULT_BROKERS
                  value: RELEASE-NAME-kafka:9092
              envFrom:
                - secretRef:
                    name: RELEASE-NAME-sentry-snuba-env
              image: getsentry/snuba:23.5.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /
                  port: 1218
                  scheme: HTTP
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 2
              name: sentry-snuba
              ports:
                - containerPort: 1218
              readinessProbe:
                failureThreshold: 10
                httpGet:
                  path: /
                  port: 1218
                  scheme: HTTP
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 2
              volumeMounts:
                - mountPath: /etc/snuba
                  name: config
                  readOnly: true
          dnsPolicy: test
          volumes:
            - configMap:
                name: RELEASE-NAME-sentry-snuba
              name: config
update image:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: sentry
        chart: sentry-19.2.0
        heritage: Helm
        release: RELEASE-NAME
      name: RELEASE-NAME-sentry-snuba-api
    spec:
      replicas: 1
      revisionHistoryLimit: 10
      selector:
        matchLabels:
          app: RELEASE-NAME-sentry
          release: RELEASE-NAME
          role: snuba-api
      template:
        metadata:
          annotations:
            checksum/config.yaml: 23f36c63e6ee38f6cd97e6889384bb53bc60cbfe742fd1c202f24d275363e850
            checksum/snubaSettingsPy: d5f85a6a8afbc55eebe23801e1a51a0fb4c0428c9a73ef6708d8dc83e079cd49
          labels:
            app: RELEASE-NAME-sentry
            release: RELEASE-NAME
            role: snuba-api
        spec:
          containers:
            - env:
                - name: SNUBA_SETTINGS
                  value: /etc/snuba/settings.py
                - name: DEFAULT_BROKERS
                  value: RELEASE-NAME-kafka:9092
              envFrom:
                - secretRef:
                    name: RELEASE-NAME-sentry-snuba-env
              image: getsentry/snuba:test
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /
                  port: 1218
                  scheme: HTTP
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 2
              name: sentry-snuba
              ports:
                - containerPort: 1218
              readinessProbe:
                failureThreshold: 10
                httpGet:
                  path: /
                  port: 1218
                  scheme: HTTP
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 2
              volumeMounts:
                - mountPath: /etc/snuba
                  name: config
                  readOnly: true
          volumes:
            - configMap:
                name: RELEASE-NAME-sentry-snuba
              name: config
update revisionHistoryLimit:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: sentry
        chart: sentry-19.2.0
        heritage: Helm
        release: RELEASE-NAME
      name: RELEASE-NAME-sentry-snuba-api
    spec:
      replicas: 1
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app: RELEASE-NAME-sentry
          release: RELEASE-NAME
          role: snuba-api
      template:
        metadata:
          annotations:
            checksum/config.yaml: 23f36c63e6ee38f6cd97e6889384bb53bc60cbfe742fd1c202f24d275363e850
            checksum/snubaSettingsPy: d5f85a6a8afbc55eebe23801e1a51a0fb4c0428c9a73ef6708d8dc83e079cd49
          labels:
            app: RELEASE-NAME-sentry
            release: RELEASE-NAME
            role: snuba-api
        spec:
          containers:
            - env:
                - name: SNUBA_SETTINGS
                  value: /etc/snuba/settings.py
                - name: DEFAULT_BROKERS
                  value: RELEASE-NAME-kafka:9092
              envFrom:
                - secretRef:
                    name: RELEASE-NAME-sentry-snuba-env
              image: getsentry/snuba:23.5.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 5
                httpGet:
                  path: /
                  port: 1218
                  scheme: HTTP
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 2
              name: sentry-snuba
              ports:
                - containerPort: 1218
              readinessProbe:
                failureThreshold: 10
                httpGet:
                  path: /
                  port: 1218
                  scheme: HTTP
                initialDelaySeconds: 10
                periodSeconds: 10
                successThreshold: 1
                timeoutSeconds: 2
              volumeMounts:
                - mountPath: /etc/snuba
                  name: config
                  readOnly: true
          volumes:
            - configMap:
                name: RELEASE-NAME-sentry-snuba
              name: config
